testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/security/jboss-ejb3.xml

```diff
@@ -17,20 +17,20 @@
         <iiop:iiop>
             <ejb-name>*</ejb-name>
             <iiop:ior-security-config>
-                <iiop:transport-config>
```

Comment:
@ropalka is this change from child elements to attributes needed? If so, it does not look like the 12 upgrade is backwards compatible

Reply:
Right the upgrade is not backward compatible @kabir .

First it brings in refactorings of few metadata packages (due to better JDK9+ support).
Second (as side effect) it brings in the fix of wrong parser implementation for IIOP transport configuration.

These configs were violating XML schema document for years. 

The question is what is the right way to handle this backward compatibility problem:
[1] Either XML Schema has precedence over wrong parser (parser impl. must be fixed).
[2] Or invalid XML parser has precedence over XML Schema specification.
IMHO [1] should be the right way how to deal with it (this is what this PR is partly about).

Just FYI this PR is follow up on this PR: https://github.com/jboss/metadata/pull/96
@ctomc & @jamezp could you also comment on this pull request, please?

Reply:
@ropalka ctomc is no longer with us :) I see @stuartwdouglas has some commits in there too though, so perhaps he can help

Reply:
@ropalka I think we will break user's applications unless the support for the 'wrong' parsing is reintroduced?

Reply:
I still maintain we need to either revert that PR or allow the XML to be parsed even if it's technically invalid. We can't break compatibility. This is especially true as I would guess most applications using IIOP are legacy applications.

Reply:
+1 to what James said. The parser needs to be modified to accept both so compatibility is preserved. There is no way this is going to be accepted otherwise, as it breaks our compatibility rules.

If we really need the JDK9 stuff then either that commit must be reverted, or someone needs to fix it.

Reply:
This was discussed quite a bit after change was done. AFAIR it was agreed that parser needs to b e added to handle the "wrong" old format as well. It was @wolfc 's team that took the fix over, but I don't know how far if anywhere it got.